### PR TITLE
Make compliant with the REUSE specification

### PR DESCRIPTION
**Pull request by Schneegans**
This makes CosmoScout VR compliant with the [REUSE Specification of the Free Software Foundation Europe](https://reuse.software/).
* It is now clear for each and every file
  * Who is the copyright holder?
  * What is the license for the file?
* There's a new CI job which enforces this.
* There's a new badge in the README to show the status.

~~This requires #298 to be merged first.~~

Some additional changes:
* The Vive controller model was removed due to the unclear licensing.
* Some moon textures have been removed due to unclear licenses.
* Some `eslint` files were removed as we are using `clang-format` to format the JavaScript files.

### CODE REVIEW

**Comment by JonasGilg** [comment]
Is there a guideline, which file is licensed under which license? Because some config files are CC and others are MIT.

**Comment by Schneegans** [comment]
> Is there a guideline for which file is licensed under which license? Because some config files are CC and others are MIT.

There should be no config files under CC-BY, however there are some under CC0.

There is no real guideline on when to choose which. However, there are two reasons due to which we may decide to publish a specific file under CC0 rather than MIT:
* The file is extremely simple. This is the case for our `.gitignore` file etc. These files do not meet a certain "threshold of originality" and are therefore not really copyrightable (see [here](https://reuse.software/faq/#what-is-copyrightable)).
* The file is very likely to be adapted by the user. For example, our scene configuration file will likely be the basis for a scene configuration file of a user. If we put it under the MIT, the user will always need to specify this which would b a pretty useless burden.

Therefore, I put very simple files and our configuration files (not the vista xml's, they actually contain complex logic) under the CC0.

File: plugins/csp-simple-bodies/textures/callisto.jpg.license [hunk 1]
```diff
@@ -0,0 +1,3 @@
+SPDX-FileCopyrightText: 2022 John van Vliet
+
+SPDX-License-Identifier: CC-BY-4.0
```
Comment (JonasGilg) [inline plugins/csp-simple-bodies/textures/callisto.jpg.license:3]: Add newline at end of file.

```suggestion
SPDX-License-Identifier: CC-BY-4.0

```

File: plugins/csp-simple-bodies/textures/ceres.jpg.license [hunk 2]
```diff
@@ -0,0 +1,3 @@
+SPDX-FileCopyrightText: 2022 USGS Astrogeology Science Center
+
+SPDX-License-Identifier: CC0-1.0
```
Comment (JonasGilg) [inline plugins/csp-simple-bodies/textures/ceres.jpg.license:3]:
```suggestion
SPDX-License-Identifier: CC0-1.0

```

File: plugins/csp-simple-bodies/textures/charon.jpg.license [hunk 3]
```diff
@@ -0,0 +1,3 @@
+SPDX-FileCopyrightText: 2022 USGS Astrogeology Science Center
+
+SPDX-License-Identifier: CC0-1.0
```
Comment (JonasGilg) [inline plugins/csp-simple-bodies/textures/charon.jpg.license:3]:
```suggestion
SPDX-License-Identifier: CC0-1.0

```

File: plugins/csp-simple-bodies/textures/deimos.jpg.license [hunk 4]
```diff
@@ -0,0 +1,3 @@
+SPDX-FileCopyrightText: 2022 Stooke. P., Stooke Small Bodies Maps V3.0. MULTI-SA-MULTI-6-STOOKEMAPS-V3.0. NASA Planetary Data System, 2015
+
+SPDX-License-Identifier: CC0-1.0
```
Comment (JonasGilg) [inline plugins/csp-simple-bodies/textures/deimos.jpg.license:3]:
```suggestion
SPDX-License-Identifier: CC0-1.0

```

File: plugins/csp-simple-bodies/textures/dione.jpg.license [hunk 5]
```diff
@@ -0,0 +1,3 @@
+SPDX-FileCopyrightText: 2022 NASA
+
+SPDX-License-Identifier: CC0-1.0
```
Comment (JonasGilg) [inline plugins/csp-simple-bodies/textures/dione.jpg.license:3]:
```suggestion
SPDX-License-Identifier: CC0-1.0

```

File: plugins/csp-simple-bodies/textures/earth.jpg.license [hunk 6]
```diff
@@ -0,0 +1,3 @@
+SPDX-FileCopyrightText: 2022 NASA
+
+SPDX-License-Identifier: CC0-1.0
```
Comment (JonasGilg) [inline plugins/csp-simple-bodies/textures/earth.jpg.license:3]:
```suggestion
SPDX-License-Identifier: CC0-1.0

```

File: plugins/csp-simple-bodies/textures/enceladus.jpg.license [hunk 7]
```diff
@@ -0,0 +1,3 @@
+SPDX-FileCopyrightText: 2022 NASA
+
+SPDX-License-Identifier: CC0-1.0
```
Comment (JonasGilg) [inline plugins/csp-simple-bodies/textures/enceladus.jpg.license:3]:
```suggestion
SPDX-License-Identifier: CC0-1.0

```

File: plugins/csp-simple-bodies/textures/europa.jpg.license [hunk 8]
```diff
@@ -0,0 +1,3 @@
+SPDX-FileCopyrightText: 2022 John van Vliet
+
+SPDX-License-Identifier: CC-BY-4.0
```
Comment (JonasGilg) [inline plugins/csp-simple-bodies/textures/europa.jpg.license:3]:
```suggestion
SPDX-License-Identifier: CC-BY-4.0

```

File: plugins/csp-simple-bodies/textures/ganymede.jpg.license [hunk 9]
```diff
@@ -0,0 +1,3 @@
+SPDX-FileCopyrightText: 2022 Hermes
+
+SPDX-License-Identifier: CC0-1.0
```
Comment (JonasGilg) [inline plugins/csp-simple-bodies/textures/ganymede.jpg.license:3]:
```suggestion
SPDX-License-Identifier: CC0-1.0

```

File: plugins/csp-simple-bodies/textures/iapetus.jpg.license [hunk 10]
```diff
@@ -0,0 +1,3 @@
+SPDX-FileCopyrightText: 2022 NASA
+
+SPDX-License-Identifier: CC0-1.0
```
Comment (JonasGilg) [inline plugins/csp-simple-bodies/textures/iapetus.jpg.license:3]:
```suggestion
SPDX-License-Identifier: CC0-1.0

```

File: plugins/csp-simple-bodies/textures/io.jpg.license [hunk 11]
```diff
@@ -0,0 +1,3 @@
+SPDX-FileCopyrightText: 2022 USGS Astrogeology Science Center
+
+SPDX-License-Identifier: CC0-1.0
```
Comment (JonasGilg) [inline plugins/csp-simple-bodies/textures/io.jpg.license:3]:
```suggestion
SPDX-License-Identifier: CC0-1.0

```

File: plugins/csp-simple-bodies/textures/jupiter.jpg.license [hunk 12]
```diff
@@ -0,0 +1,3 @@
+SPDX-FileCopyrightText: 2022 NASA / JPL-Caltech / SSI / Southwest Research Institute / Malin Space Science Systems / Italian Space Agency (ASI) / Italian National Institute for Astrophysics (INAF) / JIRAM / Björn Jónsson
+
+SPDX-License-Identifier: CC-BY-3.0
```
Comment (JonasGilg) [inline plugins/csp-simple-bodies/textures/jupiter.jpg.license:3]:
```suggestion
SPDX-License-Identifier: CC-BY-3.0

```

File: plugins/csp-simple-bodies/textures/mimas.jpg.license [hunk 13]
```diff
@@ -0,0 +1,3 @@
+SPDX-FileCopyrightText: 2022 NASA
+
+SPDX-License-Identifier: CC0-1.0
```
Comment (JonasGilg) [inline plugins/csp-simple-bodies/textures/mimas.jpg.license:3]:
```suggestion
SPDX-License-Identifier: CC0-1.0

```

File: plugins/csp-simple-bodies/textures/venus.jpg.license [hunk 14]
```diff
@@ -0,0 +1,3 @@
+SPDX-FileCopyrightText: 2022 SolarSystemScope Authors
+
+SPDX-License-Identifier: CC-BY-4.0
```
Comment (JonasGilg) [inline plugins/csp-simple-bodies/textures/venus.jpg.license:3]:
```suggestion
SPDX-License-Identifier: CC-BY-4.0

```

File: plugins/csp-simple-bodies/textures/vesta.jpg.license [hunk 15]
```diff
@@ -0,0 +1,3 @@
+SPDX-FileCopyrightText: 2022 USGS Astrogeology Science Center
+
+SPDX-License-Identifier: CC0-1.0
```
Comment (JonasGilg) [inline plugins/csp-simple-bodies/textures/vesta.jpg.license:3]:
```suggestion
SPDX-License-Identifier: CC0-1.0

```

File: plugins/csp-wms-overlays/gui/wms_settings.html [hunk 16]
```diff
@@ -39,4 +44,4 @@
   <div class="col-7">
     <div data-callback="wmsOverlays.setPrefetchCount"></div>
   </div>
-</div>
+</div>
```
Comment (JonasGilg) [inline plugins/csp-wms-overlays/gui/wms_settings.html:47]:
```suggestion
</div>

```

File: test/cs-utils/filesystem.cpp [hunk 17]
```diff
@@ -20,26 +21,26 @@ TEST_CASE("cs::utils::filesystem::create_and_list_dir") {
...
   std::set<std::string> result = cs::utils::filesystem::listFiles("./testDir");
- 
+
   CHECK(result.size() == 2);
   CHECK_EQ(*result.begin(), "./testDir/testfile.tmp");
   CHECK_EQ(*(++result.begin()), "./testDir/testfile.txt");
 };
 
-
 } // namespace cs::utils
```
Comment (JonasGilg) [inline test/cs-utils/filesystem.cpp:46]:
```suggestion
} // namespace cs::utils

```

**Review by JonasGilg — CHANGES_REQUESTED** [review]
Most change requests are adding new lines at the end if .license files.

File: tools/New-Plugin.ps1 [hunk 18]
```diff
@@ -145,10 +146,11 @@ function New-CMakeLists($directory) {
     $cMakeListsText = @"
 # ------------------------------------------------------------------------------------------------ #
 #                                This file is part of CosmoScout VR                                #
-#       and may be used under the terms of the MIT license. See the LICENSE file for details.      #
-#                         Copyright: (c) 2019 German Aerospace Center (DLR)                        #
 # ------------------------------------------------------------------------------------------------ #
 
+# SPDX-FileCopyrightText: German Aerospace Center (DLR) <cosmoscout@dlr.de>
+# SPDX-License-Identifier: MIT
+
```
[the remark this answers is not in the document]
Comment (JonasGilg) [reply tools/New-Plugin.ps1:153]: Maybe create a test plugin to check if all files get the correct headers.

**Comment by Schneegans** [comment]
Why do you want to have newlines in those files? They are auto-generated via `reuse addheader --copyright="German Aerospace Center (DLR) <cosmoscout@dlr.de>" --license="CC0-1.0" <binary file>`. It would be a bit cumbersome if we want to have those newlines...
`

**Comment by JonasGilg** [comment]
> Thank you for the review! Why do we want to have newlines in those files? They are auto-generated anyways (via `reuse addheader --copyright="German Aerospace Center (DLR) <cosmoscout@dlr.de>" --license="CC0-1.0" <binary file>`). I think it would be a bit cumbersome if we want to have those newlines everywhere...

Ok, I think we can leave it then. GitHub suggested these changes, so I went with it.

**Review by JonasGilg — APPROVED** [review]

**Pull request merged**